### PR TITLE
Fix bug where unexpected queuing  occurs in `qml.ctrl` among other functions

### PR DESCRIPTION
**Pull request by KetpuntoG**
The issue was that unwanted operators were being added when applying the `qml.ctrl`, `qml.prod`, `qml.cond` and `qml.adjoint` functions if they were located as arguments or kwargs.

To solve it I do a recursive search, for operators inside args and kwargs and I remove them.

## Drawback:

This will delete it always. I can't think of a case where we want to add an operator to the queue by passing it as an argument.

Fixes https://github.com/PennyLaneAI/pennylane/issues/6267
[sc-73690]

### CODE REVIEW

File: pennylane/ops/op_math/prod.py [hunk 1]
```diff
@@ -111,6 +111,7 @@ def prod(*ops, id=None, lazy=True):
         @wraps(fn)
         def wrapper(*args, **kwargs):
             qs = qml.tape.make_qscript(fn)(*args, **kwargs)
+
```
Comment (KetpuntoG) [inline pennylane/ops/op_math/prod.py:114]:
```suggestion
```

File: pennylane/ops/op_math/controlled.py [hunk 2]
```diff
@@ -199,11 +199,29 @@ def create_controlled_op(op, control, control_values=None, work_wires=None):
     return _ctrl_transform(op, control, control_values, work_wires)
 
 
+def remove_from_queue_args_and_kwargs(item):
```
Comment (astralcai) [inline pennylane/ops/op_math/controlled.py:202]: Maybe move this helper method to the `tape` module and just call it something like `recursively_remove_operators_from_queue`.

**Review by astralcai — COMMENTED** [review]
`qml.cond` also contains similar logic. Can you also update that to use this recursive method?

File: pennylane/ops/op_math/controlled.py (hunk 2 above)
Comment (KetpuntoG) [reply pennylane/ops/op_math/controlled.py:202]: Sounds good, is okey in `tape/operation_recorder.py`?

**Comment by KetpuntoG** [comment]
> `qml.cond` also contains similar logic. Can you also update that to use this recursive method?

I tried this but it seems `qml.cond` only accept operators, not functions as `qml.ctrl` or `qml.adjoint`

File: pennylane/ops/op_math/controlled.py (hunk 2 above)
Comment (astralcai) [reply pennylane/ops/op_math/controlled.py:202]: Might be more suitable in `pennylane/queueing.py` actually.

Comment (albi3ro) [reply pennylane/ops/op_math/controlled.py:202]: We can use pytrees and simply do:
```
leaves, _ = qml.pytrees.flatten((args, kwargs), lambda obj: isinstance(obj, Operator))
_ = [qml.QueuingManager.remove(l) for l in leaves if isinstance(l, Operator)]
```

Given this is just two lines, I'd be worried that adding a public function adds more complexity and coupling that it solves. Duplicate code is not a bad thing when it helps keep modules simple and loosely coupled.

I'd prefer to just copy those two lines into `adjoint`, `controlled`, and `TransformDispatcher._qfunc_transform`.

**Pull request closed without merging**